lib/core/src/sdk.rs

```diff
@@ -2771,6 +2774,22 @@ impl LiquidSdk {
                 // An unconfirmed tx that was not found in the payments table
                 self.persister.insert_or_update_payment_with_wallet_tx(tx)?;
             }
+            unconfirmed_txs_by_id.remove(&tx_id);
+        }
+
+        for unknown_unconfirmed_tx_data in unconfirmed_txs_by_id.values() {
+            if unknown_unconfirmed_tx_data.timestamp.is_some_and(|t| {
+                (utils::now().saturating_sub(t)) > NETWORK_PROPAGATION_GRACE_PERIOD.as_secs() as u32
+            }) {
+                self.persister
+                    .delete_payment_tx_data(&unknown_unconfirmed_tx_data.tx_id)?;
+                info!(
+                    "Found an unknown unconfirmed tx and deleted it. Txid: {}",
+                    unknown_unconfirmed_tx_data.tx_id
+                );
+            } else {
+                debug!("Found an unknown unconfirmed tx. Keeping it to allow propagation through the network. Txid: {}", unknown_unconfirmed_tx_data.tx_id)
+            }
```

Comment:
Do you think it is better to run it in the track_new_blocks loop right after the wallet full_scan ?
This way we are sure the wallet is synced before we delete the transaction while here if the full_scan haven't been running for a while (network issues) we could end up deleting an existing transaction.

Also in this case we may reduce the grace period even to 20-30 seconds IMO since it is only the time for the sdk to see that tx in the mempool.

Reply:
Maybe I misunderstood your comment, but it already runs in that exact place: `sync_payments_with_chain_data` < `sync` < `track_new_blocks`.

In my experiments, the wallet is not seeing the tx until it confirms, so 20-30 seconds wouldn't be enough. I proposed 3 minutes because it's large enough that accidental deletions are fully prevented. We could reduce it, but IMO it's fine to have a slightly higher delay given the low likelihood of this correction being actually needed.

Reply:
After discussing out-of-band with @roeierez, we concluded there was a potential issue where if `sync()` is called by the user and if the last full scan is old (e.g. after the app is backgrounded for a while), a tx may be deleted unintentionally. Moving the full scan into the sync method prevents this issue: [12ad903](https://github.com/breez/breez-sdk-liquid/pull/731/commits/12ad903d77d82e2589fd46ae38560275d99ff118)